pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -180,44 +180,47 @@ func TestCreatePod(t *testing.T) {
 	}
 
 	for _, item := range testCases {
-		ctx, cancel := context.WithCancel(context.Background())
-		fakeClientset := fake.NewSimpleClientset()
-		controller, podIndexer, _ := setupNewNoExecuteTaintManager(ctx, fakeClientset)
-		controller.recorder = testutil.NewFakeRecorder()
-		go controller.Run(ctx)
-		controller.taintedNodes = item.taintedNodes
-
-		podIndexer.Add(item.pod)
-		controller.PodUpdated(nil, item.pod)
-		// wait a bit
-		time.Sleep(timeForControllerToProgress)
-
-		podDeleted := false
-		for _, action := range fakeClientset.Actions() {
-			if action.GetVerb() == "delete" && action.GetResource().Resource == "pods" {
-				podDeleted = true
+		t.Run(item.description, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			fakeClientset := fake.NewSimpleClientset()
+			controller, podIndexer, _ := setupNewNoExecuteTaintManager(ctx, fakeClientset)
```

Comment:
actually, it doesn't look like NewNoExecuteTaintManager requires a context. Please remove it.

Reply:
I have revised it according to your comments, can you help me review it again, thank you